projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js

```diff
@@ -106,6 +106,39 @@ export default function ResumableUpload( { file } ) {
 
 	const fileSizeLabel = filesize( file.size );
 
+	const getErrorMessage = () => {
+		let errorMessage = __(
+			'An error was encountered during the upload. Check your network connection.',
+			'jetpack'
+		);
+		if ( typeof error === 'object' ) {
+			const apiResponse = error.toString().match( /message":"([^"]+)"/ );
+			// tus doesnt give us direct acces to the API response, but let's try to parse it to provide useful feedback for the user.
+			if ( typeof apiResponse === 'object' && apiResponse.length === 2 ) {
+				const apiResponseMessage = apiResponse[ 1 ];
+				// Let's give this error a better message.
+				if ( apiResponseMessage === 'Invalid Mime' ) {
+					errorMessage = (
+						<>
+							{ __( 'The format of the video you uploaded is not supported.', 'jetpack' ) }
+							&nbsp;
+							<a
+								href="https://wordpress.com/support/videopress/recommended-video-settings/"
+								target="_blank"
+								rel="noreferrer"
+							>
+								{ __( 'Check the recommended video settings.', 'jetpack' ) }
+							</a>
+						</>
+					);
+				} else {
```

Comment:
Would it be possible to add another case here for the free video upload used string, and link it to the upgrade page? We could link the `Upgrade to a VideoPress plan` text like:

[Upgrade to a VideoPress plan](#) to unlock more videos and 1TB of storage.

Reply:
Yeah, I thought about it but the PR was getting big enough. Also, I wanted to have some feedback in this regex approach before moving forward with more things